File: Web/backend/python-web-app/basic-server.py
from http.server import SimpleHTTPRequestHandler, HTTPServer

hostName = "localhost"
serverPort = 8080

class MyServer(SimpleHTTPRequestHandler):
    def do_GET(self):
        self.send_response(200)
        self.send_header("Content-type", "text/html")
        self.end_headers()
        self.wfile.write(bytes("<html><head><title>Basic Python Web Server</title></head>", "utf-8"))
        self.wfile.write(bytes("<body><h1>Hello, World!</h1></body></html>", "utf-8"))

if __name__ == "__main__":
    webServer = HTTPServer((hostName, serverPort), MyServer)
    print(f"Server started http://{hostName}:{serverPort}")

    try:
        webServer.serve_forever()
    except KeyboardInterrupt:
        pass

    webServer.server_close()
    print("Server stopped.")

File: Web/backend/ruby-web-app/basic_server.rb
require 'webrick'

server = WEBrick::HTTPServer.new(:Port => 8080)

server.mount_proc '/' do |req, res|
  res.body = '<html><head><title>Basic Ruby Web Server</title></head><body><h1>Hello, World!</h1></body></html>'
end

trap 'INT' do
  server.shutdown
end

server.start

